### PR TITLE
APIs

### PR DESCRIPTION
**Pull request by ng-anhhtuann**

### CODE REVIEW

File: src/main/java/com/example/GHTK/Repository/Repository.java [hunk 1]
```diff
@@ -223,10 +225,171 @@ public Object insert(Type type) throws SQLException {
...
+        return object;
+    }
+
+    public Object queryTime() {
+        Object object;
+        List<TimeLine> timeList = new ArrayList<>();
+
+        String query = "SELECT * FROM GHTK.Time;";
+        try {
+            Connection connection = DriverManager.getConnection(url, username, password);
```
Comment (uranashel44) [inline src/main/java/com/example/GHTK/Repository/Repository.java:278]: Same

File: src/main/java/com/example/GHTK/Repository/Repository.java [hunk 2]
```diff
@@ -223,10 +225,171 @@ public Object insert(Type type) throws SQLException {
...
+        }
+        return object;
+    }
+
+    public Object queryAreaById(String idArea) {
+        Object object = null;
+        Area area = null;
+        String query = "SELECT * FROM GHTK.Area WHERE idArea ='" + idArea + "';";
+        try {
+            Connection connection = DriverManager.getConnection(url, username, password);
```
Comment (uranashel44) [inline src/main/java/com/example/GHTK/Repository/Repository.java:257]: Same as above

File: src/main/java/com/example/GHTK/Repository/Repository.java [hunk 3]
```diff
@@ -223,10 +225,171 @@ public Object insert(Type type) throws SQLException {
...
+        return object;
+    }
+
+    public Object queryService() {
+        Object object = null;
+        List<Service> serviceList = new ArrayList<>();
+
+        String query = "SELECT * FROM GHTK.Service;";
+        try {
+            Connection connection = DriverManager.getConnection(url, username, password);
```
Comment (uranashel44) [inline src/main/java/com/example/GHTK/Repository/Repository.java:318]: Same

File: src/main/java/com/example/GHTK/Repository/Repository.java [hunk 4]
```diff
@@ -246,21 +409,22 @@ public Object queryCustomer(){
...
     }
-    public Object queryCustomerById(String idCustomer){
+
+    public Object queryCustomerById(String id) {
         Object object = null;
         Customer customer = null;
-        String query = "SELECT * FROM GHTK.Customer WHERE idCustomer ='"+idCustomer+"';";
+        String query = "SELECT * FROM GHTK.Customer WHERE idCustomer ='" + id + "';";
         try {
             Connection connection = DriverManager.getConnection(url, username, password);
```
Comment (uranashel44) [inline src/main/java/com/example/GHTK/Repository/Repository.java:424]: Same

**Review by uranashel44 — CHANGES_REQUESTED** [review]
Some comments need to be updated, it is mandatory

File: src/main/java/com/example/GHTK/Repository/Repository.java [hunk 5]
```diff
@@ -223,10 +225,171 @@ public Object insert(Type type) throws SQLException {
         }
         return object;
     }
+
     /*
-    Get customer's information by id or get all
+    GET API (rights only) for admin account
      */
-    public Object queryCustomer(){
+    public Object queryArea() {
```
[the remark this answers is not in the document]
Comment (ng-anhhtuann) [reply src/main/java/com/example/GHTK/Repository/Repository.java:232]: SQLException will be thrown on time per request to show users what's wrong with the input data they send.
SQLException is required in connection._getConnection()_

**Review by uranashel44 — APPROVED** [review]

File: src/main/java/com/example/GHTK/Ultilities/Database.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package com.example.GHTK.Ultilities;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class Database {
```
Comment (uranashel44) [inline src/main/java/com/example/GHTK/Ultilities/Database.java:10]: It it singleton?

**Review by uranashel44 — APPROVED** [review]

**Pull request merged**